### PR TITLE
Fix location ids and building catalogue

### LINKED ISSUE #856: fix: location_ids and buildingCatalogue are inconsistent
Continuing the saga from #784 and #779, `location_ids` and `buildingCatalogue` are inconsistent.

Noted in #855:
> Not all locations are recognized for custom events on the map even though they are for classes (BS3, for example) and some don't even read on the calendar

In the particular case above,

buildingCatalogue:
```ts
    83057: {
        name: 'Biological Sciences III (BS3)',
        lat: 33.645363,
        lng: -117.846222,
        imageURLs: ['463/biological-sciences-iii.jpg'],
    },

    114877: {
        name: 'Biological Sciences III (BS3)',
        lat: 33.645363,
        lng: -117.846222,
        imageURLs: ['463/biological-sciences-iii.jpg'],
    },
```

location_ids:
```ts
    BS3: 114877,
```

Custom Events seemingly use the 83057 which don't exist in `location_ids`, whereas course events use 114877 which do.

### PR DESCRIPTION
**Pull request by srukelman**
Created 1:1 parity between location ids and buildings catalogue

closes #856

### CODE REVIEW

File: apps/antalmanac/tests/locations.ts
Comment (KevinWu098) [inline apps/antalmanac/tests/locations.ts:1]: This file should be renamed to `locations.test.ts`, otherwise it won't be detected as a test

**Review by KevinWu098 — CHANGES_REQUESTED** [review]
Good work so far. However, this doesn't pass the test cases Brian wrote, as the file was misnamed so it wasn't actually being run.

Along the lines of not passing the tests, there still exists duplicates in `buildingCatalogue.ts`, such as "Biological Sciences III (BS3)", which exists under `83057` and `114877`.

>  FAIL  |0| tests/locations.test.ts > buildingCatalogue > no duplicate names
AssertionError: expected 318 to be 297 // Object.is equality

>  FAIL  |0| tests/locations.test.ts > locationIds and buildingCatalogue > all locationIds are in buildingCatalogue
AssertionError: expected [ 'Art Studio (ART)', …(317) ] to include 'SCS'

>  FAIL  |0| tests/locations.test.ts > locationIds and buildingCatalogue > all buildingCatalogue are in locationIds
AssertionError: expected [ 'SCS', 'MPS', 'GC', 'CD', …(293) ] to include 'Art Studio (ART)'

File: apps/antalmanac/tests/locations.ts
Comment (ap0nia) [reply apps/antalmanac/tests/locations.ts:1]: Lol, that's my fault

**Review by KevinWu098 — APPROVED** [review]
LGTM 🚀 , merge when you're able @srukelman

**Pull request merged**